### PR TITLE
Jetpack Compose

### PR DESCRIPTION
**Pull request by pocmo**
The first small steps towards Jetpack Compose support.

Adding the "do not land" label until automation is green.

### CODE REVIEW

**Comment by pocmo** [comment]
Building the UI test docker image failed. Looks like it couldn't download flank? Wrong URL?

```
INFO[0053] Running: [/bin/sh -c URL_FLANK_BIN=$(curl -s "https://api.github.com/repos/Flank/flank/releases" | grep "browser_download_url*" | grep "${FLANK_VERSION}" | sed -r "s/\"//g" | cut -d ":" -f3)     && wget "https:${URL_FLANK_BIN}" -O ${TEST_TOOLS}/flank.jar     && chmod +x ${TEST_TOOLS}/flank.jar] 
--2021-05-26 09:54:31--  ftp://https/
           => ‘.listing’
Resolving https (https)... failed: Name or service not known.
wget: unable to resolve host address ‘https’
error building image: error building stage: failed to execute command: waiting for process to exit: exit status 4
```

**Comment by pocmo** [comment]
Some steps further.

Currently the tests are a problem:
> 1609 tests completed, 201 failed, 10 skipped

I upgraded mockk, but there are still some failing. Almost all of them seem to be:

> io.mockk.MockKException: Class cast exception happened.
Probably type information was erased.
In this case use `hint` before call to specify exact return type of a method.

**Comment by pocmo** [comment]
Okay, this is a bit stupid... but it seems like this makes this specific test pass:

```diff
diff --git a/app/src/test/java/org/mozilla/fenix/HomeActivityTest.kt b/app/src/test/java/org/mozilla/fenix/HomeActivityTest.kt
index 9c086a9f5..8cc48456d 100644
--- a/app/src/test/java/org/mozilla/fenix/HomeActivityTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/HomeActivityTest.kt
@@ -105,7 +105,10 @@ class HomeActivityTest {

         val settings: Settings = mockk()
         every { settings.shouldReturnToBrowser } returns true
-        every { activity.components.settings.shouldReturnToBrowser } returns true
+        every {
+            hint(FenixApplication::class)
+            activity.components.settings.shouldReturnToBrowser
+        } returns true
         every { activity.openToBrowser(any(), any()) } returns Unit

         activity.browsingModeManager = browsingModeManager
```

**Comment by pocmo** [comment]
I had to disable 5 unit tests and filed a follow-up issue for looking at those: https://github.com/mozilla-mobile/fenix/issues/20506

File: app/src/test/java/org/mozilla/fenix/shortcut/PwaOnboardingObserverTest.kt [hunk 1]
```diff
@@ -52,13 +61,13 @@ class PwaOnboardingObserverTest {
         settings = mockk(relaxed = true)
         webAppUseCases = mockk(relaxed = true)
 
-        pwaOnboardingObserver = spyk(PwaOnboardingObserver(
+        pwaOnboardingObserver = PwaOnboardingObserver(
```
Comment (csadilek) [inline app/src/test/java/org/mozilla/fenix/shortcut/PwaOnboardingObserverTest.kt:64]: Pushed these changes to fix tests and "unignored" them. :)

**Comment by csadilek** [comment]
@pocmo pushed more updates, all tests pass locally for me now.

**Review by csadilek — APPROVED** [review]
⬆️ 11

**Comment by pocmo** [comment]
Mergify can't merge this PR because a GitHub workflow was modified. Merging manually.

**Pull request merged**